main.tf

```diff
@@ -33,20 +33,26 @@ resource "aws_security_group" "internal" {
 
   tags = "${merge(var.tags, map("Name", var.cluster_name,
                                 "Cluster", var.cluster_name))}"
+}
 
-  ingress {
-    from_port   = 0
-    to_port     = 0
-    protocol    = "-1"
-    cidr_blocks = ["${var.subnet_range}"]
-  }
+resource "aws_security_group_rule" "internal_ingress_rule" {
```

Comment:
@bernadinm lets document here why this change is needed. Can you please explain why we have to change from the previous all in one resource?

as `var.subnet_range` is a array and the user needs to know his networks beforehand anyways why do we have to introduce this breaking change.

Is the whole resource getting recreated on a change in this variable?

Reply:
I've described it in detail here: terraform-providers/terraform-provider-aws#7058

but the TL;DR is that when you rerun `terraform apply` terraform thinks that everything that is in the `aws_security_group` should be there and nothing else. Which breaks when you add new regions to that internal group. Terraform will notice it and will remove it during the next terraform apply, and they will keep going back and forth of adding, and removing, etc.